src/snowflake/snowpark/_internal/utils.py

```diff
@@ -423,14 +444,19 @@ def calculate_checksum(
                 # ignore generated python files
                 if ignore_generated_py_file and file.endswith(GENERATED_PY_FILE_EXT):
                     continue
+
                 hash_algo.update(file.encode("utf8"))
-                if current_size < chunk_size:
-                    filename = os.path.join(dirname, file)
-                    file_size = os.path.getsize(filename)
+
+                filename = os.path.join(dirname, file)
+                file_size = os.path.getsize(filename)
+
+                if whole_file_hash:
+                    _hash_file(hash_algo, filename, chunk_size, whole_file_hash)
+                    current_size += file_size
+                elif current_size < chunk_size:
                     read_size = min(file_size, chunk_size - current_size)
                     current_size += read_size
```

Comment:
reading the logic, actually is `current_size` used after set?

Reply:
Yes, on future iterations `current_size` is used to see if this branch should be entered.

Reply:
ahh I see, that code is unexpanded in the review page